src/core/xlio.h

```diff
@@ -382,6 +384,225 @@ int xlio_socketxtreme_ref_buff(struct xlio_buff_t *buff);
  */
 int xlio_socketxtreme_free_buff(struct xlio_buff_t *buff);
 
+/*
+ * Storage API
+ */
+
+typedef uintptr_t xlio_group_t;
+typedef uintptr_t xlio_socket_t;
+typedef uint32_t xlio_key_t;
+
+struct xlio_buf {
+    uint64_t userdata;
+};
+
+/*
+ * XLIO initialization.
+ *
+ * xlio_init_ex() must be called before using any storage API. This is heavy operation.
+ * xlio_init_ex() is not thread-safe operation, however, subsequent serialized calls exit
+ * successfully without any action.
+ *
+ * If set, memory_cb() notifies about memory blocks which zerocopy RX buffers can point to.
+ * Current implementation allocates a single memory block and does it within xlio_init_ex() context.
+ */
+
+/*
+ * Memory callback.
+ *
+ * XLIO calls the callback each time XLIO allocates a memory region which can be used for RX
+ * buffers. User can use this information to prepare the memory for some logic in the future.
+ * Zerocopy RX interface provides pointers to such memory.
+ *
+ * Current XLIO implementation does a single allocation for buffers.
+ */
+typedef void (*xlio_memory_cb_t)(void *addr, size_t len, size_t hugepage_size);
+
+struct xlio_init_attr {
+    unsigned flags;
```

Comment:
Consider sing something that can easily be 'grep'd without huge duplicates in the code, like init_flags or anything that can be distiguished in simple text search.

Reply:
I will leave it as is because:
1. The attribute structure describes specific object/subsystem and it's unambiguous as result
2. Simple field "flags" follows the well known APIs (like POSIX), so this also follows least surprise principle
3. This is public API and shouldn't go deep into XLIO implementation, therefore, the common word "flags" shouldn't make big impact on the text grep.